File: GithubUserJava/app/src/main/java/com/example/myapplication/ui/FollowFragment.java
package com.example.myapplication.ui;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;
import androidx.recyclerview.widget.LinearLayoutManager;

import com.example.myapplication.adapter.UserAdapter;
import com.example.myapplication.databinding.FragmentFollowBinding;
import com.example.myapplication.data.remote.response.ItemsItem;
import com.example.myapplication.factory.FollowViewModelFactory;
import com.example.myapplication.viewmodel.FollowViewModel;

import java.util.List;
import java.util.stream.Collectors;

public class FollowFragment extends Fragment {

    private FragmentFollowBinding binding;

    public static final String ARG_POSITION = "position";
    public static final String ARG_USERNAME = "username";

    private int position = 0;
    private String username = "";

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        binding = FragmentFollowBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        FollowViewModel mainViewModel = new ViewModelProvider(this, new FollowViewModelFactory(username)).get(FollowViewModel.class);
        LinearLayoutManager layoutManager = new LinearLayoutManager(requireActivity());
        binding.rvFollow.setLayoutManager(layoutManager);

        mainViewModel.follow.observe(getViewLifecycleOwner(), this::setDataFollow);
        mainViewModel.isLoading.observe(requireActivity(), this::showLoading);

        Bundle arguments = getArguments();
        if (arguments != null) {
            position = arguments.getInt(ARG_POSITION);
            username = arguments.getString(ARG_USERNAME, "");
        }

        if (position == 1) {
            mainViewModel.findUserFollowers(username);
        } else {
            mainViewModel.findUserFollowing(username);
        }
    }

    private void setDataFollow(List<ItemsItem> dataUserFollow) {
        List<ItemsItem> getUserData = dataUserFollow.stream()
                .map(item -> new ItemsItem(item.getLogin(), item.getAvatarUrl()))
                .collect(Collectors.toList());

        UserAdapter adapter = new UserAdapter(getUserData);
        binding.rvFollow.setAdapter(adapter);
    }

    private void showLoading(boolean isLoading) {
        binding.progressBar.setVisibility(isLoading ? View.VISIBLE : View.GONE);
    }
}

File: GithubUserKotlin/app/src/main/java/com/example/myapplication/ui/FollowFragment.kt
package com.example.myapplication.ui


import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.viewModels
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.myapplication.adapter.UserAdapter
import com.example.myapplication.databinding.FragmentFollowBinding
import com.example.myapplication.data.remote.response.ItemsItem
import com.example.myapplication.factory.FollowViewModelFactory
import com.example.myapplication.viewmodel.FollowViewModel


class FollowFragment : Fragment() {

    private lateinit var binding: FragmentFollowBinding

    companion object {
        const val ARG_POSITION = "position"
        const val ARG_USERNAME = "username"
    }

    private var position = 0
    private var username = ""

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {

        binding = FragmentFollowBinding.inflate(inflater,container,false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        val mainViewModel : FollowViewModel by viewModels {
            FollowViewModelFactory(username)
        }
        val layoutManager = LinearLayoutManager(requireActivity())
        binding.rvFollow.layoutManager = layoutManager
        mainViewModel.follow.observe(viewLifecycleOwner) {follow -> setDataFollow(follow)}
        mainViewModel.isLoading.observe(requireActivity()) {showLoading(it)}
        arguments?.let {
            position = it.getInt(ARG_POSITION)
            username = it.getString(ARG_USERNAME).toString()
        }
        if (position == 1) {
            mainViewModel.findUserFollowers(username)
        } else {
            mainViewModel.findUserFollowing(username)
        }
    }

    private fun setDataFollow(dataUserFollow : List<ItemsItem>) {
        val getUserData = dataUserFollow.map {
            ItemsItem(it.login, it.avatarUrl)
        }
        val adapter = UserAdapter(getUserData)
        binding.rvFollow.adapter = adapter
    }
    private fun showLoading(isLoading: Boolean) {
        binding.progressBar.visibility = if (isLoading) View.VISIBLE else View.GONE
    }
}